contribute/process/commit_template

```diff
@@ -0,0 +1,16 @@
+# <50 character summary of the change
```

Comment:
This is the most critical part.
Less is More. However it critical.
The reader will read this as if this commit is check-ed in it will <50 char commit summary >

Give couple of sample answers.

Reply:
Sample examples in commit templates?

Reply:
I meant less than 50 characters

Reply:
Removing the first line, "<50 character...".
Most people will anyway add the first line of the commit, so deleting the first line.

Reply:
I just read [the first stackoverflow answer](https://softwareengineering.stackexchange.com/questions/42110/can-you-recommend-a-good-commit-message-template-guidelines-to-enforce-in-the) here. Looks good. Shall we proceed with this one?

Reply:
This is what looks like in my laptop:
```
amit:e2e$ cat /home/amit/.gitmessage 
```

```
some awesome title

1. Why is this change necessary ?

* 

2. How does this change address the issue ?

* 

3. How to verify this change ?

*

4. What side effects does this change have ?

*

```

Reply:
We cannot remove the title
We have to explain how a title is read.
I was trying to explain you just that.
Chat with @satyamz for some links regarding how a commit title is read

Reply:
@AmitKumarDas This file has been moved to `.github` directory as prateek suggested

Reply:
Title of the commit message should be imperative and should be wrapped up in 50 chars not more than that. 
For example: You should read commit message as, 

If this commit is applied then this commit will `Add new parameter to template`
If this commit is applied then this commit will `Fix parameters in the template`
If this commit is applied then this commit will `Improve the code structure`

Title of the commit message would be part `highlighted`

It should be **_imperative_** , Please keep these things in mind and provide same info somewhere so that people will learn from this. 
Thanks